addons/website/models/website.py

```diff
@@ -1848,36 +1841,36 @@ def _trigram_enumerate_words(self, search_details, search, limit):
                 and not self.env.user.has_group('base.group_user')
             )
             if filter_is_published:
-                where_clause = sql.SQL("WHERE is_published")
+                where_clause = SQL("WHERE is_published")
 
-            from_clause = sql.SQL("FROM {table}").format(table=sql.Identifier(model._table))
+            from_clause = SQL("FROM %(table)s", table=SQL.identifier(model._table))
             # Specific handling for fields being actually part of another model
             # through the `inherits` mechanism.
             for table_to_join in {
                 field['table']: field['fname'] for field in inherits_fields.values()
             }.items():  # Removes duplicate inherits model
-                from_clause = sql.SQL("""
-                    {from_clause}
-                    LEFT JOIN {inherits_table} ON {table}.{inherits_field} = {inherits_table}.id
-                """).format(
-                    from_clause=from_clause,
-                    table=sql.Identifier(model._table),
-                    inherits_table=sql.Identifier(table_to_join[0]),
-                    inherits_field=sql.Identifier(table_to_join[1]),
-                )
-            query = sql.SQL("""
-                SELECT {table}.id, {best_similarity} AS _best_similarity
-                {from_clause}
-                {where_clause}
+                from_clause = SQL("""
+                    %(from_clause)s
+                    LEFT JOIN %(inherits_table)s ON %(table)s.%(inherits_field)s = %(inherits_table)s.id
+                """,
+                                  from_clause=from_clause,
+                                  table=SQL.identifier(model._table),
+                                  inherits_table=SQL.identifier(table_to_join[0]),
+                                  inherits_field=SQL.identifier(table_to_join[1]),
+                                  )
+            query = SQL("""
+                SELECT %(table)s.id, %(best_similarity)s AS _best_similarity
+                %(from_clause)s
+                %(where_clause)s
```

Comment:
OMG I am pretty sure this could be refactored with `_field_to_sql()` and letting the latter generate the necessary JOINs.  But that's not the purpose of this PR.